Library/Homebrew/diagnostic.rb

```diff
@@ -443,6 +443,25 @@ def check_access_cellar
         EOS
       end
 
+      def check_for_empty_taps
+        taps_dir = HOMEBREW_LIBRARY/"Taps"
+        return unless taps_dir.exist?
+        return unless taps_dir.directory?
+
+        found_empty_dirs = []
+
+        # Warn USER of empty taps_dir children directories
+        taps_dir.children.select{ |c| c.directory? }          \
+          .select { |c| (Dir.entries(c) - %w[. ..]).empty? }  \
+          .each { |c| found_empty_dirs.push(c) }
+
+        return unless found_empty_dirs.length > 0
+
+        <<-EOS.undent
+          Found #{found_empty_dirs.length} empty directories in your Taps directory located in #{HOMEBREW_LIBRARY}. This has been known to cause `brew update` to report updated and new formulae multiple times. To fix this issue consider removing the empty folders manually or by running `cd $(brew --prefix)/Library/Taps && rmdir *`.
```

Comment:
Sorry, this was probably my fault. When I said not to worry about wrapping to 80 characters in doctor-related stuff, I meant the code itself rather than the user-facing bits _(the printed warnings)_. Anything user-facing should still be wrapped.